app/models/pawn.rb

```diff
@@ -1,15 +1,17 @@
 # Pawn behavior.
 class Pawn < Piece
   # Returns true if the pawn made a valid upgraded forward
-  # move on its first move, or else a regular forward move.
-  # Capture, collision, check, and checkmate logic are not
-  # implemented yet and thus ignored.
+  # move on its first move, or else a regular forward move,
+  # and ensure no capture takes place by moving forward.
+  # Check, and checkmate logic are not implemented yet
+  # and thus ignored.
   def valid_move?(x, y)
-    moved ? one_forward_move?(x, y) : first_forward_move?(x, y)
+    (moved ? one_forward_move?(x, y) : first_forward_move?(x, y)) &&
+      !forward_capture?(x, y)
```

Comment:
What happens if there is an enemy directly in front of pawn on its first move, and pawn attempts a forward +2 move?